### ISSUE TITLE
Item mixin to return changed item count has excessive overhead

### ISSUE THREAD

**Issue by Linguardium** [newhoryzon/farmers-delight-fabric#175]
I would suggest, alternatively, modifying the Item itself. 

since maxCount is simply a field, just create an accessor and set it on all stew items currently in the registry, then listen to the registry using RegistryEntryAddedCallback.

This loses the ability to dynamically turn it on, but gains the fact that no processing is happening on the fly _every time an item count is looked up_

**Comment by Zifiv** [comment]
Hello @Linguardium,

Thanks you for submitting this open point.
Do you have any report (from spark plugin/mod for example) that show it's excessive overhead ?

I've tried some things about modifying directly the `maxCount` and it works well, but it need more changes, because if the configuration is changed, (for example disabling the feature) we should come back to the previous value of `maxCount` so I need to store previous value.

It is interesting and I'll keep the ticket open but isn't my priority for now.

**Comment by Linguardium** [comment]
ultimately all the processing you do when getMaxCount is called, will be called for any mod trying to transfer items. so any block entity for instances doing a check if it can insert and item will cause your logic to fire, to include the required processing just to enter the method and return if it isnt a stew item. it happens for any item. So in a modded environment where you have pipe systems, all that extra processing and object creation/destruction is quite a lot compared to a primitive field return.

**Comment by Zifiv** [comment]
> As mentioned, you would lose the ability to hotswap the configuration. you would need to restart the game to restore (or store the original value in a map of some sort)

It will not lose the ability to hotswap, I've already done the dev to do with the hotswap config.

**Comment by Linguardium** [comment]
a CallbackInfoReturnable is created each time and the return value is checked every time as well. its not just your code but the wrapping mixin code that is generated and processed as well.

**Comment by Technici4n** [comment]
See
![image](https://github.com/newhoryzon/farmers-delight-fabric/assets/13494793/df23e11c-795c-4d37-8e2b-fd98e843934c)

(from https://spark.lucko.me/Fsml1FqUXj).

**Comment by vectorwing** [comment]
Yeah, this is a rather old implementation... After asking for some help regarding this, I decided to remove the `maxStackSize()` part of the mixin, and use reflection on game load to change those values.

Due to this change, I'm also potentially removing the `overrideAllSoupItems` config, as I would need to scan the entire registry to apply this reflection to relevant items. I might ship without this function, and rethink it later.

**Comment by Linguardium** [comment]
Fabric has hooks for registry additions and gives you the reference to the item. Accessors will let you set this value directly while it is being registered. As I mentioned previously you can iterate the item registry and modify on your mod initializer and then register a callback to check the ones registered after
